contracts/src/CrossChainApplications/NativeTokenBridge/README.md

```diff
@@ -0,0 +1,34 @@
+# Native-to-Native Token Bridge
+
+A pair of smart contracts built on top of Teleporter to support using an ERC20 token or the native token of any `subnet-evm` chain as the native token for a given subnet.
+
+## Design
+The native-to-native bridge is implemented using two primary contracts.
+- `NativeTokenSource`
+    - Lives on the `Source chain`. Pairs with exactly one `NativeTokenDestination` contract on a different chain.
+    - Locks and unlocks native tokens on the Source chain corresponding to mints and burns on the destination chain.
+    - `transferToDestination`: transfers all tokens payed to this function call to `recipient` on the destination chain by locking them and instructing the destination chain to mint. Optionally takes the address of an ERC20 contract `feeContractAddress` as well as an amount `feeAmount` that will be used as the relayer-incentivisation for the teleporter cross-chain call. Also allows for the caller to specify `allowedRelayerAddresses`.
+    - `receiveTeleporterMessage`: unlocks tokens on the source chain when instructed to by the `NativeTokenDestination` contract.
+- `NativeTokenDestination`
+    - Lives on the `Destination chain`. Pairs with exactly one `NativeTokenSource` contract on a different chain.
+    - Mints and burns native tokens on the Destination chain corresponding to locks and unlocks on the source chain.
+    - `transferToSource`: transfers all tokens payed to this function call to `recipient` on the source chain by burning the tokens and instructing the source chain to unlock. Optionally takes the address of an ERC20 contract `feeContractAddress` as well as an amount `feeAmount` that will be used as the relayer-incentivisation for the teleporter cross-chain call. Also allows for the caller to specify `allowedRelayerAddresses`.
+    - `receiveTeleporterMessage`: mints tokens on the destination chain when instructed to by the `NativeTokenDestination` contract.
+    - `isCollateralized`: returns true if `currentReserveImbalance == 0`, meaning that enough tokens have been sent from the source chain to offset the `initialReserveImbalance`. If true, all tokens sent to the destination chain will be minted, and burning/unlocking tokens will be enabled.
+    - `totalSupply`: Returns the best estimate of available native tokens on this chain. Equal to the `initialReserveImbalance` + `all tokens minted` - `all tokens in known burn address`. Known burn addresses include the burn address for this contract used when burning/unlocking, and the address for burned transaction fees.
+
+- `Collateralizing the bridge`
```

Comment:
Let's make this as well as the below top-level bullet points ### subsections instead